### ISSUE TITLE
useColors() method causes ReferenceError: 'navigator' is not defined

### ISSUE THREAD

**Issue by sstone1** [visionmedia/debug#374]
The following code when used in a JavaScript interpreter for Go ([Otto](https://github.com/robertkrimen/otto/tree/master/otto)) throws an error:  
`ReferenceError: 'navigator' is not defined`.
This worked fine in previous versions of the module (I've tested 2.3.3).
Could a check be added to see if `navigator` exists?

```javascript
function useColors() {
  // NB: In an Electron preload script, document will be defined but not fully
  // initialized. Since we know we're in Chrome, we'll just detect this case
  // explicitly
  if (typeof window !== 'undefined' && 'process' in window && window.process.type === 'renderer') {
    return true;
  }

  // is webkit? http://stackoverflow.com/a/16459606/376773
  // document is undefined in react-native: https://github.com/facebook/react-native/pull/1632
  return (typeof document !== 'undefined' && 'WebkitAppearance' in document.documentElement.style) ||
    // is firebug? http://stackoverflow.com/a/398120/376773
    (typeof window !== 'undefined' && window.console && (console.firebug || (console.exception && console.table))) ||
    // is firefox >= v31?
    // https://developer.mozilla.org/en-US/docs/Tools/Web_Console#Styling_messages
    (navigator.userAgent && navigator.userAgent.toLowerCase().match(/firefox\/(\d+)/) && parseInt(RegExp.$1, 10) >= 31) ||
    // double check webkit in userAgent just in case we are in a worker
    (navigator.userAgent && navigator.userAgent.toLowerCase().match(/applewebkit\/(\d+)/));
}
```

**Comment by thebigredgeek** [comment]
@sstone1 this was fixed yesterday in 2.4.3

**Comment by sstone1** [comment]
@thebigredgeek the problem is occurring with debug 2.4.4 (I've just retested to confirm).
The problem is occurring because `navigator` is undefined in this JavaScript interpreter
I think it needs a `typeof navigator !== 'undefined'` test before the `navigator.userAgent` test.

**Comment by sstone1** [comment]
Looks like @jochenberger has got a fix for this in PR #376
